torchao/_models/llama/eval.py

```diff
@@ -94,6 +96,23 @@ def run_evaluation(
             quantizer = Int4WeightOnlyGPTQQuantizer(groupsize=groupsize, device=device)
             model.setup_caches(max_batch_size=1, max_seq_length=calibration_seq_length)
             model = quantizer.quantize(model, inputs).to(device)
+        elif "awq" in quantization:
+            from torchao.prototype.awq.api import ObservedLinear, insert_awq_observer, awq_quant 
+            insert_awq_observer(model, precision, device)
```

Comment:
nit: `insert_awq_observer` --> `insert_awq_observer_` to indicate it modifies the model inplace?

Reply:
also I'd suggest to make `awq_quant` a function as well to align with other configs like `int4_weight_only()`

Reply:
I think just `awq()`  seems to be general enough with work with uintx and floatx? if so we can just use `awq()` I think. otherwise maybe `awq_uintx()`

Reply:
yea I think awq_uintx() for now. I haven't tested awq more generally